public/app/controllers/mainCtrl.js

```diff
@@ -40,63 +37,25 @@ angular.module('kibibitCodeEditor')
         controller: 'projectFolderModalController',
         controllerAs: 'projectFolderModalCtrl',
         className: 'ngdialog-theme-default',
-        scope: $scope
+        scope: $scope,
+        data: {
+          userHomeDirectoryPath: vm.userHomeDirectoryPath
+        }
+      }).closePromise.then(function(selectedProjectFolder) {
+        EventManagerService.trigger(
```

Comment:
not sure I understood your comment. I asked what this does. It seems like this updates the `fileTree` with data from the `mainCtrl`.

I need some explaining on what this do so I'll know why this is needed before you ask me if I want you to do some change for something I don't understand